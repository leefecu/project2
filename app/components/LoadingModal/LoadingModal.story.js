import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';

import LoadingModal from './LoadingModal';
import Styles from '../../Stories.styles';

storiesOf('Component | LoadingModal', module).add('Primary', () => {
  return (
    <View style={Styles.containerWithBG}>
      <View style={Styles.row}>
        <Text style={Styles.title}>HUH</Text>
        <LoadingModal />
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>OOPS</Text>
        <LoadingModal />
      </View>
    </View>
  );
});
