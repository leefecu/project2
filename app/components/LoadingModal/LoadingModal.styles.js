import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

const SMALL_DEVICE = Metrics.width <= 320; // iphone 5

export default StyleSheet.create({
  header: {
    position: 'relative',
    alignItems: 'center',
  },
  headerLine: {
    position: 'absolute',
    backgroundColor: Colors.primaryDark,
    height: 1,
    top: 9,
    width: Metrics.width - Metrics.doubleMargin,
  },
  headerTextWrapper: {
    backgroundColor: Colors.white,
    paddingHorizontal: Metrics.doublePadding,
    zIndex: 1,
  },
  headerText: {
    color: Colors.primaryDark,
    fontSize: Metrics.Fonts.small,
    fontFamily: 'CircularPro-Bold',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  content: {
    marginVertical: SMALL_DEVICE ? Metrics.doubleMargin * 2 : Metrics.doubleMargin * 2,
  },
  message: {
    color: Colors.primaryDark,
    fontSize: SMALL_DEVICE ? Metrics.Fonts.small : Metrics.Fonts.normal,
    fontFamily: 'CircularPro-Book',
    marginBottom: SMALL_DEVICE ? Metrics.smallMargin : Metrics.doubleMargin,
    textAlign: 'center',
  },
  messageContainer: {
    marginBottom: Metrics.doubleMargin,
    paddingHorizontal: Metrics.basePadding,
  },
  loader: {
    alignItems: 'center',
  },
});
