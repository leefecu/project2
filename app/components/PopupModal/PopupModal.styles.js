import { StyleSheet } from 'react-native';

import Colors from '../../themes/Colors';
import Metrics from '../../themes/Metrics';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: Colors.modalBGColor,
    flex: 1,
    justifyContent: 'center',
  },
  innerContainer: {
    backgroundColor: Colors.white,
    borderColor: Colors.white,
    borderRadius: Metrics.inputBorderRadius,
    borderWidth: Metrics.borderWidth,
    paddingVertical: Metrics.doublePadding,
    paddingTop: Metrics.basePadding,
    paddingBottom: Metrics.doublePadding,
    paddingHorizontal: Metrics.smallerPadding,
    width: Metrics.width - Metrics.baseMargin,
  },
});
