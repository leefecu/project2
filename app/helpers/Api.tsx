import get from 'lodash/get';
import reject from 'lodash/reject';
import Config from 'react-native-config';
import compareVersions from 'compare-versions';
import { Platform } from 'react-native';

import * as Storage from './Storage';
import { createExpiryNotification } from './Notification';
import { convertStashToActivePAck } from './Transformation';
import API from '../services/Api';

import { SYSTEM_RESPONSE } from '../constants/Shared';

const api = API.create();

export const purchasePack = async (packetId: string, promoId?: string) => {
  const response = await api.purchasePack(packetId, promoId);
  const currentActivePack = await Storage.getActivePack();
  if (response.ok) {
    const balance = get(response, 'data.balance');
    const purchasedPack = get(response, 'data.purchasedPack');
    const activePack = currentActivePack
      ? {
          ...currentActivePack,
          endTime: purchasedPack.endTime,
        }
      : purchasedPack;

    createExpiryNotification(purchasedPack.endTime);
    await Storage.setActivePackAndBalance(activePack, balance);
    return response;
  }
  return response;
};

export const activateStash = async (stash: TStash) => {
  const response = await api.activateStash(stash.packetInstanceId);
  if (response.ok) {
    const currentActivePack = await Storage.getActivePack();
    const currentStash = await Storage.getStash();
    const newActivePack = convertStashToActivePAck(stash, currentActivePack);
    const newStash = reject(
      currentStash,
      (target: TStash) => target.packetInstanceId === stash.packetInstanceId,
    );

    createExpiryNotification(newActivePack.endTime);
    await Storage.saveActivePackAndStash(newActivePack, newStash);
    return response;
  }
  return response;
};

export const checkSystemInfo = async (): Promise<SYSTEM_RESPONSE> => {
  const response = await api.getSystemInfo();
  if (response.ok) {
    const maintenance: boolean = get(response, 'data.maintenance', false);
    if (maintenance) {
      return SYSTEM_RESPONSE.MAINTENANCE;
    }
    const minVersion = get(response, `data.${Platform.OS}`);
    const appVersion = Config.APP_VERSION;

    return compareVersions.compare(appVersion, minVersion, '>=')
      ? SYSTEM_RESPONSE.CONTINUE
      : SYSTEM_RESPONSE.UPGRADE;
  }
  return SYSTEM_RESPONSE.CONTINUE;
};
