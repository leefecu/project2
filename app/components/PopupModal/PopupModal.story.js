import React from 'react';
import { Text, View } from 'react-native';

import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import PopupModal from './PopupModal';
import ErrorModal from '../ErrorModal/ErrorModal';
import LoadingModal from '../LoadingModal/LoadingModal';
import Styles from '../../Stories.styles';

storiesOf('Component | PopupModal', module).add('Primary', () => {
  return (
    <View style={Styles.containerWithBG}>
      <View style={Styles.row}>
        <Text style={Styles.title}>Error</Text>
        <PopupModal visible={true}>
          <ErrorModal
            closeButtonLabel="Go back"
            headerLabel="HUH?!"
            message="Huh?! Something is wrong with that number. Please try again."
            onClose={action('close')}
          />
        </PopupModal>
      </View>
      <View style={Styles.row}>
        <Text style={Styles.title}>Loading</Text>
        <PopupModal visible={true}>
          <LoadingModal />
        </PopupModal>
      </View>
    </View>
  );
});
