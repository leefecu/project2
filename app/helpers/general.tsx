import { NavigationParams, NavigationScreenProp, NavigationState } from 'react-navigation';

import { ERROR_CODE } from '../constants/Messages';
import { SYSTEM_RESPONSE } from '../constants/Shared';

import * as API from '../helpers/Api';

export const noop = () => {};

interface TTestProp {
  testID: string;
  accessibilityLabel: string;
}
export const testProps = (id: string): TTestProp => ({
  testID: id,
  accessibilityLabel: id,
});

export const shouldLogout = (status: number | undefined, errorCode: ERROR_CODE): boolean =>
  status === 401 ||
  status === 403 ||
  errorCode === ERROR_CODE.NOT_REGISTERED ||
  errorCode === ERROR_CODE.INVALID_PLAN ||
  errorCode === ERROR_CODE.QD_ISSUE;

export const systemCheck = async (
  navigation: NavigationScreenProp<NavigationState, NavigationParams>,
) => {
  const systemInfo = await API.checkSystemInfo();
  if (systemInfo === SYSTEM_RESPONSE.CONTINUE) {
    return;
  }
  navigation.navigate(systemInfo === SYSTEM_RESPONSE.MAINTENANCE ? 'Maintenance' : 'Upgrade');
};
