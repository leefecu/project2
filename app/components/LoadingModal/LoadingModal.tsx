import React from 'react';
import { ActivityIndicator, Text, View } from 'react-native';

import { testProps } from '../../helpers/general';

import Styles from './LoadingModal.styles';

import Colors from '../../themes/Colors';

const LoadingModal = () => (
  <View {...testProps('Loading Modal')}>
    <View style={Styles.header}>
      <View style={Styles.headerLine} />
      <View style={Styles.headerTextWrapper}>
        <Text style={Styles.headerText}>LOADING</Text>
      </View>
    </View>

    <View style={Styles.content}>
      <View style={Styles.messageContainer}>
        <Text style={Styles.message}>Weâ€™re processing your request.</Text>
      </View>

      <View style={Styles.loader}>
        <ActivityIndicator size="large" color={Colors.loadingBlue} />
      </View>
    </View>
  </View>
);

export default LoadingModal;
