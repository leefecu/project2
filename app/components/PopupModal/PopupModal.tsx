import React from 'react';
import { Modal, View } from 'react-native';

import { noop, testProps } from '../../helpers/general';

import Styles from './PopupModal.styles';

interface Props {
  children?: React.ReactNode;
  visible: boolean;
}

const PopupModal = ({ children, visible }: Props) => {
  return (
    <Modal animationType="fade" transparent={true} visible={visible} {...testProps('Popup Modal')}>
      <View style={Styles.container}>
        <View style={Styles.innerContainer}>{children}</View>
      </View>
    </Modal>
  );
};

PopupModal.defaultProps = {
  onClose: noop,
  visible: false,
};

export default PopupModal;
