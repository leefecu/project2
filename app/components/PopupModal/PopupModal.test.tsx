/**
 * @jest-environment jsdom
 */
import React from 'react';
import { create } from 'react-test-renderer';

import PopupModal from './PopupModal';
import LoadingModal from '../LoadingModal/LoadingModal';

import { noop } from '../../helpers/general';

describe('LoadingModal component', () => {
  test('render default LoadingModal', () => {
    const button = create(
      <PopupModal onClose={noop} visible={true}>
        <LoadingModal />
      </PopupModal>,
    );
    expect(button.toJSON()).toMatchSnapshot();
  });
});
