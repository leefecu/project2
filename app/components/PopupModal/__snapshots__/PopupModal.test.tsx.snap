// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingModal component render default LoadingModal 1`] = `
<Modal
  accessibilityLabel="Popup Modal"
  animationType="fade"
  hardwareAccelerated={false}
  testID="Popup Modal"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.7)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#ffffff",
          "borderRadius": 5,
          "borderWidth": 1,
          "paddingBottom": 24,
          "paddingHorizontal": 8,
          "paddingTop": 16,
          "paddingVertical": 24,
          "width": 734,
        }
      }
    >
      <View
        accessibilityLabel="Loading Modal"
        testID="Loading Modal"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "position": "relative",
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "#004071",
                "height": 1,
                "position": "absolute",
                "top": 9,
                "width": 718,
              }
            }
          />
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "paddingHorizontal": 24,
                "zIndex": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004071",
                  "fontFamily": "CircularPro-Bold",
                  "fontSize": 17,
                  "fontWeight": "bold",
                  "textAlign": "center",
                }
              }
            >
              LOADING
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginVertical": 64,
            }
          }
        >
          <View
            style={
              Object {
                "marginBottom": 32,
                "paddingHorizontal": 16,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#004071",
                  "fontFamily": "CircularPro-Book",
                  "fontSize": 20,
                  "marginBottom": 32,
                  "textAlign": "center",
                }
              }
            >
              Weâ€™re processing your request.
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
              }
            }
          >
            <ActivityIndicator
              animating={true}
              color="#0faae5"
              hidesWhenStopped={true}
              size="large"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
